packer/ansible/roles/isc-dhcp-server/files/config_isc-dhcp-server.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -e
+
+##########################################################
+# retrieve NIC name via ```ip addr``` command, by given id
+##########################################################
+get_nic_name_by_index()
+{
+    #######################
+    # This script is to obtain and echo the NIC name (lo/eth0/eth1) by index(1/2/3) shown in ```ip addr```
+    #
+    #
+    # ip addr will show:
+    #
+    #1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default
+    #.......
+    #2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 ...
+    #.......
+    #3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pf....
+    #......
+    #
+    ########################
+    tmp=$( ip addr|grep "^${1}:" | awk '{print $2}')
+    if [[ "$tmp" == "" ]];
+    then
+        echo "[Error]: Invalid Parameter passed to get_nic_name_by_index(): NIC ID=$1";
```

Comment:
1. add numeric checking
2. add a check of the awk print output, to limit it in "xxx:" formating.